pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,250 @@
+
+    if len(features) > len(wires):
+        raise ValueError("Number of bits to embed cannot be larger than number of wires which is {}, "
+                         "got {}.".format(len(wires), len(features)))
+    BasisState(features, wires=wires)
+
+
+def SqueezingEmbedding(features, execution='amplitude', c=0.1, wires=None):
+    r"""Encodes :math:`M` features into the squeezing amplitudes :math:`r` or phases :math:`\phi` of :math:`M` modes.
+
+    The mathematical definition of the squeezing gate is given by the operator
+
+    .. math::
+
+        S(z) = \exp\left(\frac{r}{2}\left(e^{-i\phi}\a^2 -e^{i\phi}{\ad}^{2} \right) \right),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``len(wires)`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+
+    Keyword Args:
+        execution (str): ``'phase'`` encodes the input into the phase of single-mode squeezing, while
+            ``'amplitude'`` uses the amplitude
+        c (float): parameter setting the value of the phase of all squeezing gates if ``execution='amplitude'``, or the
+            amplitude of all squeezing gates if ``execution='phase'``
+        wires (Sequence[int]): sequence of mode indices that the template acts on
+    """
+
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if len(wires) != len(features):
+        raise ValueError("SqueezingEmbedding requires a feature vector of size n_wires which is {}"
+                         ", got {}.".format(len(wires), len(features)))
+
+    for f, w in zip(features, wires):
+        if execution == 'amplitude':
+            Squeezing(f, c, wires=w)
+        elif execution == 'phase':
+            Squeezing(c, f, wires=w)
+        else:
+            raise ValueError("Execution strategy {} not known. Has to be `phase` or `amplitude`.".format(execution))
+
+
+def DisplacementEmbedding(features, execution='amplitude', c=0.1, wires=None):
+    r"""Encodes :math:`M` features into the displacement amplitudes :math:`r` or phases :math:`\phi` of :math:`M` modes.
+
+    The mathematical definition of the displacement gate is given by the operator
+
+    .. math::
+            D(\alpha) = \exp(r (e^{i\phi}\ad -e^{-i\phi}\a)),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``len(wires)`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+
+    Keyword Args:
+        execution (str): ``'phase'`` encodes the input into the phase of single-mode displacement, while
+            ``'amplitude'`` uses the amplitude
+        c (float): parameter setting the value of the phase of all displacement gates if ``execution='amplitude'``, or the
```

Comment:
Same as for squeezing embedding:
parameter setting the value of the phase of all squeezing gates if **execution='phase'**, or the amplitude of all squeezing gates if **execution='amplitude'**

phase/amplitude need to be flipped
also is c=[0,1]? it will be good to mention the range here